src/browser/modules/Stream/Views/AsciiView.jsx

```diff
@@ -18,15 +18,66 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+import { Component } from 'preact'
 import asciitable from 'ascii-data-table'
+import { debounce } from 'services/utils'
+import { StyledStatsBar, PaddedDiv, StyledBodyMessage, StyledRightPartial, StyledWidthSliderContainer, StyledWidthSlider } from '../styled'
 
-import { PaddedDiv, StyledBodyMessage } from '../styled'
-
-const AsciiView = ({rows, style, message}) => {
-  const contents = rows
-    ? <pre>{asciitable.tableFromSerializedData(rows, 70)}</pre>
-    : <StyledBodyMessage>{message}</StyledBodyMessage>
+const AsciiView = ({rows, style, message, maxColWidth = 70}) => {
+  let contents = <StyledBodyMessage>{message}</StyledBodyMessage>
+  if (rows !== undefined && rows.length) {
+    contents = <pre>{asciitable.tableFromSerializedData(rows, maxColWidth)}</pre>
+  }
   return <PaddedDiv style={style}>{contents}</PaddedDiv>
 }
 
+export class AsciiStatusbar extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      maxSliderWidth: 140,
+      maxColWidth: 70
+    }
+  }
+  componentWillReceiveProps (nextProps) {
+    this.ensureContentWidth(nextProps.rows)
+  }
+  ensureContentWidth (rows) {
+    if (rows.length !== this.props.rows.length) {
+      this.setMaxSliderWidth(asciitable.maxColumnWidth(rows))
+    }
+  }
+  componentWillMount () {
+    this.debouncedMaxColWidthChanged = debounce(this.maxColWidthChanged, 25, this)
```

Comment:
With large result sets it's really expensive to recalculate the table 10 times per second and will probably crash the web browser.